boards.txt

```diff
@@ -5506,6 +5506,40 @@ Midatronics.menu.upload_method.dfuMethod.upload.protocol=2
 Midatronics.menu.upload_method.dfuMethod.upload.options=-g
 Midatronics.menu.upload_method.dfuMethod.upload.tool=stm32CubeProg
 
+################################################################################
+# elecgator boards
+
+elecgator.name=elecgator boards
+
+elecgator.build.core=arduino
+elecgator.build.board=elecgator
+elecgator.build.variant_h=variant_{build.board}.h
+
+# EtherCATduino board
+elecgator.menu.pnum.ETHERCAT_DUINO=EtherCATduino
+elecgator.menu.pnum.ETHERCAT_DUINO.node=NODE_F746ZET
+elecgator.menu.pnum.ETHERCAT_DUINO.upload.maximum_size=524288
+elecgator.menu.pnum.ETHERCAT_DUINO.upload.maximum_data_size=327680
+elecgator.menu.pnum.ETHERCAT_DUINO.build.mcu=cortex-m7
+elecgator.menu.pnum.ETHERCAT_DUINO.build.flags.fp=-mfpu=fpv4-sp-d16 -mfloat-abi=hard
+elecgator.menu.pnum.ETHERCAT_DUINO.build.board=ETHERCAT_DUINO
+elecgator.menu.pnum.ETHERCAT_DUINO.build.series=STM32F7xx
+elecgator.menu.pnum.ETHERCAT_DUINO.build.product_line=STM32F746xx
+elecgator.menu.pnum.ETHERCAT_DUINO.build.variant=STM32F7xx/F745Z(E-G)T_F746Z(E-G)(T-Y)_F750Z8T_F756ZG(T-Y)
+elecgator.menu.pnum.ETHERCAT_DUINO.build.extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial} {build.bootloader_flags}
+elecgator.menu.pnum.ETHERCAT_DUINO.build.cmsis_lib_gcc=arm_cortexM7lfsp_math
+
+# Upload menu
+elecgator.menu.upload_method.swdMethod=STM32CubeProgrammer (SWD)
+elecgator.menu.upload_method.swdMethod.upload.protocol=0
+elecgator.menu.upload_method.swdMethod.upload.options=-g
+elecgator.menu.upload_method.swdMethod.upload.tool=stm32CubeProg
+
+elecgator.menu.upload_method.dfuMethod=STM32CubeProgrammer (DFU)
+elecgator.menu.upload_method.dfuMethod.upload.protocol=2
+elecgator.menu.upload_method.dfuMethod.upload.options=-g
+elecgator.menu.upload_method.dfuMethod.upload.tool=stm32CubeProg
+
```

Comment:
Please move it to ESC boards section, as it seems to be an ESC board. we would not add a dedicated section.